apps/prairielearn/src/ee/auth/lti13/lti13Auth.ts

```diff
@@ -149,9 +158,58 @@ const OIDCAuthResponseSchema = z.object({
 router.post(
   '/callback',
   asyncHandler(async (req, res) => {
+    debug('callback');
+    debug(req.method);
+    debug(req.body);
+
+    // https://www.imsglobal.org/spec/security/v1p0/#step-3-authentication-response
+    OIDCAuthResponseSchema.passthrough().parse(req.body);
+
     const lti13_instance = await selectLti13Instance(req.params.lti13_instance_id);
 
-    const lti13_claims = await authenticate(req, res);
+    const openidClientConfig = await getOpenidClientConfig(lti13_instance);
+
+    // Needed for implicit flow
+    client.useIdTokenResponseType(openidClientConfig);
+
+    // URL href doesn't matter, openid-client only uses the url.hash to pass properties
+    // into client.implicitAuthentication
+    const url = new URL('https://example.com/');
+    url.hash = new URLSearchParams(req.body).toString();
```

Comment:
This is gross 😬 seems like it's just coercing to `URLSearchParams` internally anyways. I may open a PR to allow it to take `URLSearchParams` directly. While it's true that today they only look at the hash, we have no guarantee of that in the future.